web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -113,15 +113,17 @@ export const ConnectMyComputerContextProvider: FC<{
     isAgentConfiguredAttempt.data;
 
   const rootCluster = clustersService.findCluster(props.rootClusterUri);
-  const canUse = useMemo(
-    () =>
-      canUseConnectMyComputer(
-        rootCluster,
-        configService,
-        mainProcessClient.getRuntimeSettings()
-      ) || isAgentConfigured,
-    [configService, isAgentConfigured, mainProcessClient, rootCluster]
-  );
+  const canUse = useMemo(() => {
+    const isFeatureFlagEnabled = configService.get(
+      'feature.connectMyComputer'
+    ).value;
+    const hasPermissions = hasConnectMyComputerPermissions(
+      rootCluster,
+      mainProcessClient.getRuntimeSettings()
+    );
+
+    return isFeatureFlagEnabled && (hasPermissions || isAgentConfigured);
```

Comment:
We could note here what we said in the RFD, that the user should always have access to the agent after configuring it and link to relevant section of the RFD.